### DISCUSSION TITLE
WithErrorCallback function is not able to access the original request

### DISCUSSION THREAD

**Discussion by tobiaschielens** [corazawaf/coraza#1186] [Q&A]
We want to create an overview of blocked requests. The database for this overview is located on another server.  We want to log headers and data to that second server. 

With the `WithErrorCallback(handleErrorCallback)` we can get the reason why the request was blocked but we would also like to be able to forward more details to our second server. 

How would we be able to achieve this?

**Comment by jcchavezs** [comment]
WithErrorCallback can now also access the context.Context of the request. You can propagate details there. See https://github.com/corazawaf/coraza/blob/681787ef1267fde8affb0ebd691b38fab140bd7a/internal/corazarules/rule_match.go#L152-L179

**Reply by tobiaschielens** [reply]
We also want to access the headers and body of the original request. The context only gives us the matched rule as well.

**Reply by jcchavezs** [reply]
Since we inherit the context from the request you can write a middleware to put the request into the context in a custom middleware before the coraza middleware. We had a few attempts to allow hooks that could achieve similar stuff
- https://github.com/corazawaf/coraza/pull/987
- https://github.com/corazawaf/coraza/pull/986

However we did not proceed because we did not have serious request. One step towards that would be that you build that custom middleware and proof the value of such feature.
